packages/superset-ui-connection/src/callApi/callApi.ts

```diff
@@ -26,6 +30,41 @@ export default function callApi({
     signal,
   };
 
+  if (method === 'GET' && cacheAvailable) {
+    return caches.open('superset').then(supersetCache =>
+      supersetCache
+        .match(url)
+        .then(cachedResponse => {
+          if (cachedResponse) {
+            // if we have a cached response, send its ETag in the
+            // `If-None-Match` header in a conditional request
+            const etag = cachedResponse.headers.get('Etag');
+            if (etag) {
+              request.headers = request.headers || {};
```

Comment:
could simplify to `request.headers = { ...request.headers, ['If-None-Match']: etag }`

Reply:
Cool, I didn't know the spread operator worked with `null`s. Will do.